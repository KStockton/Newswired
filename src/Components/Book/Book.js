import React from 'react';
import PropTypes from 'prop-types';

const Book = ({bookImage, rank, author, description, title, weeksOnList}) => {
  return (
    <article className="book-tile">
        <img className="book-img" src={bookImage} alt="book"/>
        <p className="author">{author}</p>
        <p className="book-rank">Rank: {rank}</p>
        <p className="book-weeks">{weeksOnList} weeks on list </p>
        <p className="book-title">{title}</p>
        <p className="book-description">{description}</p>
      </article>
  )
}

Book.propTypes = {
  bookImage: PropTypes.string,
  rank: PropTypes.number,
  author: PropTypes.string,
  description: PropTypes.string,
  title: PropTypes.string,
  weeksOnList: PropTypes.number
}
export default Book