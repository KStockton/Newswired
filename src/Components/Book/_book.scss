.book-tile {
  border-radius: 8px;
  padding-top: 50px;
  padding-left: 25px;
  font-size: 150%;
  transition: .2s all;
  padding-right: 25px;
}
.book-wrapper:hover .book-tile {
  opacity: .5;
  transform: scale(.98);
  box-shadow: none;
}

.book-wrapper:hover .book-tile:hover{
  filter: blur (0px);
  transform: scale(1.1);
  opacity: 1;
  box-shadow: 0 8px 20px 0 px rgba(0,0,0,0.125);
}

.book-rank {
  text-align: left;
  font-family: 'Old Standard TT', serif;
  font-size: 0.9rem;
  padding: 5px 0px;
  color:white;
}

.book-img {
  height: 267px;
  box-shadow: 3px 3px 8px 3px #a99a9a;
  border-radius: 1px;
  min-width: 172px;
}
.author {
  @include authors( 12px, left, 5px);
  font-family: 'Noto Sans', sans-serif;
  padding-bottom: 16px;
  color:white;
}
.book-weeks {
  font-size: 0.8rem;
  text-align: left;
  font-family: 'Noto Sans', sans-serif;
  text-transform: uppercase;
  padding: 7px 0px;
  color:white;
}
.book-title {
  @include titles(.9rem, 1.1875rem, 700, left, 3px);
  font-family: 'Noto Sans', sans-serif;
  color:white;
}

.book-description {
  @include tiles(left, 1rem, 6px );
  font-family: 'Old Standard TT', serif;
  color:white;

}
.book-favorite {
  display: none;
  position: relative;
  top: 34px;
  width: 25px;
  left: 66px
}

.book-nofavorite{
  display: none;
  position: relative;
  top: 34px;
  width: 25px;
  left: 66px;
}