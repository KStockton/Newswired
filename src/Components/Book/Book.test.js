import React from 'react';
import { shallow } from 'enzyme';
import {Book} from './Book';
import {Link} from 'react-router-dom';


describe('Book', () => {
  let wrapper;
  let mockProp = jest.fn();
  let mockId = 3;
  
  beforeEach(() => {
    wrapper = shallow(<Book toggleFavorite={mockProp} id={mockId}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when props are passed', () => {
    let mockbookImage = "http://win.com";
    let mockrank = 1;
    let mockauthor = "Michele Obama";
    let mockdescription = 'effort';
    let mocktitle = 'Overcome';
    let mockweeksOnList = 2;

    const wrapper = shallow(<Book
      bookImage={mockbookImage}
      rank={mockrank}
      description={mockdescription}
      author={mockauthor}
      title={mocktitle}
      weeksOnList={mockweeksOnList}
      toggleFavorite={mockProp}
    />);
    
    expect(wrapper).toMatchSnapshot();
  });

  it('Link should have a prop of /card/3 when a user clicks the link', () => {

    let mockId = 3;
    const wrapper = shallow(<Book id={mockId}/>);
    expect(wrapper.find(Link).props().to).toBe('/card/3');
  });

  it('should call togglefavorite when img is clicked', () => {
    wrapper.find('#img-test').simulate('click');
    expect(mockProp).toHaveBeenCalled();
  });

  it('toggle faavorites should be called with the it', () => {
    wrapper.find('#img-test').simulate('click');
    expect(mockProp).toHaveBeenCalledWith(3);
  });

});