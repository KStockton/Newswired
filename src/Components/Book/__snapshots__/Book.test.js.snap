// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should match the snapshot 1`] = `
<article
  className="book-tile"
>
  <img
    alt="emoji favorite icon"
    className="book-nofavorite"
    src="nolove.svg"
  />
  <img
    alt="book"
    className="book-img"
  />
  <p
    className="author"
  />
  <p
    className="book-rank"
  >
    Rank: 
  </p>
  <p
    className="book-weeks"
  >
     weeks on list
  </p>
  <p
    className="book-title"
  />
  <p
    className="book-description"
  />
</article>
`;

exports[`Book should match the snapshot when props are passed 1`] = `
<article
  className="book-tile"
>
  <img
    alt="emoji favorite icon"
    className="book-nofavorite"
    src="nolove.svg"
  />
  <img
    alt="book"
    className="book-img"
    src="http://cool.com"
  />
  <p
    className="author"
  >
    Michele Obama
  </p>
  <p
    className="book-rank"
  >
    Rank: 
    3
  </p>
  <p
    className="book-weeks"
  >
    4
     weeks on list
  </p>
  <p
    className="book-title"
  >
    Overcome
  </p>
  <p
    className="book-description"
  >
    effor
  </p>
</article>
`;
